### ISSUE TITLE
[Question] Is `typing_extensions.Self` supposed for us to use?

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pylance-release#2034]
I see that it just exists in the bundled typeshed, while it doesn't exist in the real library. Is it supposed for us to use, or something Pylance plans to use internally?

Code would look like:
```python
from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from typing_extensions import Self

class Base:
    def make_list(self) -> list[Self]:
        return [self]
```

**Comment by erictraut** [comment]
Support for `Self` is provisional, as a way to test the usability of a draft PEP. You shouldn't use it in your code at this point. If the PEP is accepted, runtime support will be added (presumably in Python 3.11 and in an upcoming version of `typing_extensions`). If the PEP is rejected, I will remove the provisional support from pyright.

Pyright already used an internal `Self` type for `self` and `cls` parameters when no explicit type annotation is provided.

**Comment by PabloLION** [comment]
Python 3.11 is out: https://docs.python.org/3.11/whatsnew/3.11.html
And the `Self` PEP https://peps.python.org/pep-0673 got accepted

Yay  🎉

**Comment by erictraut** [comment]
Yes, `Self` is fully supported now that PEP 673 has been accepted. It is available in the `typing` module for Python 3.11 and in `typing_extensions` for backward compatibility with older versions of Python.

**Comment by PabloLION** [comment]
Thanks for pointing out `typing_extensions`. 
I auto-import-ed it and the error `ModuleNotFoundError: No module named 'typing_extensions'` is gone after installing the package.
